docs/datasource_api_userguide.md

```diff
@@ -0,0 +1,271 @@
+  .option("table", "target_table_append")
+  .mode(SaveMode.Append)
+  .save()
+```
+
+### Read using spark sql
+```sql
+CREATE TABLE test1
+  USING tidb
+  OPTIONS (
+    database 'tpch_test',
+    table 'CUSTOMER',
+    tidb.addr '127.0.0.1',
+    tidb.password '',
+    tidb.port '4000',
+    tidb.user 'root',
+    spark.tispark.pd.addresses '127.0.0.1:2379',
+    spark.tispark.plan.allow_index_read 'true'
+  );
+
+select C_NAME from test1 where C_CUSTKEY = 1;
+```
+
+### Write using spark sql
+```sql
+CREATE TABLE writeUsingSparkSQLAPI_src
+  USING tidb
+  OPTIONS (
+    database 'tpch_test',
+    table 'CUSTOMER',
+    tidb.addr '127.0.0.1',
+    tidb.password '',
+    tidb.port '4000',
+    tidb.user 'root',
+    spark.tispark.pd.addresses '127.0.0.1:2379',
+    spark.tispark.plan.allow_index_read 'true'
+  );
+
+// target_table should exist in tidb
+CREATE TABLE writeUsingSparkSQLAPI_dest
+  USING tidb
+  OPTIONS (
+    database 'tpch_test',
+    table 'target_table',
+    tidb.addr '127.0.0.1',
+    tidb.password '',
+    tidb.port '4000',
+    tidb.user 'root',
+    spark.tispark.pd.addresses '127.0.0.1:2379',
+    spark.tispark.plan.allow_index_read 'true'
+  );
+
+// insert into values
+insert into writeUsingSparkSQLAPI_dest values
+     (1000,
+     "Customer#000001000",
+     "AnJ5lxtLjioClr2khl9pb8NLxG2",
+     9,
+     "19-407-425-2584",
+     2209.81,
+     "AUTOMOBILE",
+     ". even, express theodolites upo");
+
+// insert into select
+insert into writeUsingSparkSQLAPI_dest select * from writeUsingSparkSQLAPI_src;
+
+// insert overwrite select
+insert overwrite table writeUsingSparkSQLAPI_dest select * from writeUsingSparkSQLAPI_src;
+```
+
+## Using the Spark Connector With Extensions Enabled
+Let's see how to use the connector with extensions enabled.
+
+### init SparkConf
+```scala
+val sparkConf = new SparkConf()
+  .setIfMissing("spark.master", "local[*]")
+  .setIfMissing("spark.app.name", getClass.getName)
+  .setIfMissing("spark.sql.extensions", "org.apache.spark.sql.TiExtensions")
+  .setIfMissing("tidb.addr", "127.0.0.1")
+  .setIfMissing("tidb.password", "")
+  .setIfMissing("tidb.port", "4000")
+  .setIfMissing("tidb.user", "root")
+  .setIfMissing("spark.tispark.pd.addresses", "127.0.0.1:2379")
+  .setIfMissing("spark.tispark.plan.allow_index_read", "true")
+
+val spark = SparkSession.builder.config(sparkConf).getOrCreate()
+val sqlContext = spark.sqlContext
+```
+
+### Read using scala
+```scala
+// use tidb config in spark config if does not provide in data source config
+val tidbOptions: Map[String, String] = Map()
+val df = sqlContext.read
+  .format("tidb")
+  .options(tidbOptions)
+  .option("database", "tpch_test")
+  .option("table", "CUSTOMER")
+  .load()
+  .filter("C_CUSTKEY = 1")
+  .select("C_NAME")
+df.show()
+```
+
+### Write using scala
+```scala
+// use tidb config in spark config if does not provide in data source config
+val tidbOptions: Map[String, String] = Map()
```

Comment:
yes, tidb config is provided in `SparkConf`. 
We can also provide tidb config here, and it will overwrite the tidb config in `SparkConf`.
see `Use another TiDB` Section.

Reply:
Oh! So we actually does not need `tidbOptions` here?

Reply:
And add an example SparkConf perhaps?

Reply:
yes, i put `tidbOptions` here in order to let users aware that they do not need to pass any parameters.

Reply:
`SparkConf` is defined in line 174-184

Reply:
Oh I missed that, my bad.